### ISSUE TITLE
JS Module Support

### ISSUE THREAD

**Issue by smahood** [clojure/clojurescript-site#65]
The pages 
https://clojurescript.org/guides/javascript-modules 
and 
https://clojurescript.org/reference/javascript-module-support
seem to be pretty similar in both name and scope. I suspect that they should either be combined into a single page, or split into a more specific reference and guide with a little less crossover.

I'll start work on a PR if nobody beats me to it, though it may be a while - I'll need to understand things a bit better first. I think there's also been some relevant improvements in this space since the last edits as well.

**Comment by swannodette** [comment]
@smahood feel to ask any questions you might have here. Would like to see these consolidated.

**Comment by swannodette** [comment]
Yeah the reference should be less guide-like. PR welcome from anyone who's interested.
